experiment/simulation/js/main.js

```diff
@@ -1 +1,579 @@
-//Your JavaScript goes in here
+const submit_button = document.getElementById("submit");
+const next_button = document.getElementById("next-button");
+const prev_button = document.getElementById("prev-button");
+const reset_button = document.getElementById("reset-button");
+const formx1 = document.getElementById("x1");
+const formy1 = document.getElementById("y1");
+const formx2 = document.getElementById("x2");
+const formy2 = document.getElementById("y2");
+const errorx1 = document.getElementById("error-x1");
+const errorx2 = document.getElementById("error-x2");
+const errory1 = document.getElementById("error-y1");
+const errory2 = document.getElementById("error-y2");
+// get the canvas
+
+let canvas = document.getElementById("canvas");
+// get the context
+let ctx = canvas.getContext("2d");
+const height = 600;
+const width = 1200;
+canvas.height = height;
+canvas.width = width;
+
+const chosen_color = "green";
+const possible_color = "blue";
+// determine the number of divisions based on the distance between the two points
+const divisions = 45;
+const MAXX = 88;
+const MINX = -100;
+const MAXY = 44;
+const MINY = 0;
+const grid_color = "white";
+const originx = 45;
+const originy = 23;
+const block_size = height / divisions;
+
+// define the set of coordinates depict them as 2-d vectors
+let point1 = [];
+let point2 = [];
+let decision_parameter;
+let dp = [];
+// fix the heights of the canvas
+let slope = 0;
+let currx, curry;
+let dx, dy;
+let last_move_direction = "";
+let valid = 1;
+
+// let submit_button = document.getElementById("submit");
+// store the number of times next is called
+let times_next_called = 0;
+
+formx1.onchange = function () {
+  let x1 = formx1.value;
+  if (x1 < -45 || x1 > 44) {
+    valid = 0;
+    formx1.style.border = "1px solid red";
```

Comment:
Can we put this style under a css class and use the class across all the elements which need to represent erroneous input by using [classList.add/remove/toggle](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList)?